chia/wallet/wallet_transaction_store.py

```diff
@@ -152,6 +155,11 @@ async def increment_sent(
         sent_to.append(append_data)
 
         tx: TransactionRecord = dataclasses.replace(current, sent=sent_count, sent_to=sent_to)
+        if not tx.is_spendable:
+            # if the tx is not spendable, it is confirmed that we can't spend it, but
+            # we need to adjust the GUI to show that it is failed
+            log.info(f"Marking tx={tx.name} as confirmed but failed, since it is not spendable due to errors")
```

Comment:
Does TransactionRecord have a "failed" state? I thought we only had confirmed or not.

Reply:
it doesn't but I defined a failed state as having `confirmed=1 and confirmed_at_height=0` which we can use to display as failed in UX/GUI

Reply:
this feels a bit scary. it should be clearly documented, because it's not obvious from the code that we're treating this special case as such